grakn-core/src/main/java/ai/grakn/graql/Var.java

```diff
@@ -58,4 +68,33 @@
 
     @Override
     int hashCode();
+
+    /**
+     * The {@link Kind} of a {@link Var}, such as whether it is user-defined.
+     */
+    enum Kind {
```

Comment:
hahahahahaha I am guessing this is because "Type" is so overloaded in our world ? :D

Reply:
🏆